pytensor/scalar/basic.py

```diff
@@ -1598,8 +1598,8 @@ def L_op(self, inputs, outputs, gout):
         second_part = switch(cond, 0.0, gz)
 
         if outputs[0].type in discrete_types:
-            first_part = 0.0
-            second_part = 0.0
+            first_part = ift.zeros_like(config.floatX)
```

Comment:
Doesn't that usually take an array?

Reply:
This is a method of a PyTensor variable, it's equivalent to `zeros_like(self)` or something